code/modules/library/library_catalog.dm

```diff
@@ -0,0 +1,605 @@
+ * and prepping information from the database for use elsewhere
+ */
+
+///External proc that Returns a report library_category datum that matches the provided cat_id
+/datum/library_catalog/proc/getReportCategoryByID(category_id)
+	for (var/datum/library_category/category in GLOB.library_catalog.report_types)
+		if(category.category_id == category_id)
+			return category
+	//proc shouldn't get this far, but if there's an entry in the DB that we don't have added, just default to other cat
+	for (var/datum/library_category/category in GLOB.library_catalog.report_types)
+		if(category.category_id == LIB_REPORT_OTHER)
+			return category
+
+///External proc that Returns a book library_category datum that matches the provided cat_id
+/datum/library_catalog/proc/getBookCategoryByID(category_id)
+	for (var/datum/library_category/category in GLOB.library_catalog.categories)
+		if(category.category_id == category_id)
+			return category
+
+///External proc that Returns a report programmaticbook datum that matches the provided bookid
+/datum/library_catalog/proc/getProgrammaticBookByID(id)
+	for(var/book in GLOB.library_catalog.books)
+		var/datum/programmaticbook/PB = book
+		if(PB.id == id)
+			return PB
+
+/*
+  * # getBookByID
+  *
+  * External proc that takes in an id number and searches the Database for a book with a matching SSID
+  * returns a cached book with the data from that row
+  *
+  * Arguments:
+  * * id - integer value that matches as a book SSID
+ */
+/datum/library_catalog/proc/getBookByID(id)
+	var/datum/db_query/query = SSdbcore.NewQuery("SELECT id, author, title, content, summary, ratings, raters, primaryCategory, secondaryCategory, tertiaryCategory, ckey, reports FROM library WHERE id=:id", list(
+		"id" = id
+	))
+
+	if(!query.warn_execute())
+		qdel(query)
+		return
+
+	var/list/results = list()
+	while(query.NextRow())
+		var/datum/cachedbook/CB = new()
+		CB.LoadFromRow(list(
+			"id"      = query.item[1],
+			"author"  = query.item[2],
+			"title"   = query.item[3],
+			"content" = query.item[4],
+			"summary" = query.item[5],
+			"ratings" = query.item[6],
+			"raters"  = query.item[7],
+			"primaryCategory"   = query.item[8],
+			"secondaryCategory" = query.item[9],
+			"tertiaryCategory"  = query.item[10],
+			"ckey"    = query.item[11],
+			"reports" = query.item[12],
+		))
+		results += CB
+		qdel(query)
+		return CB
+	qdel(query)
+	return results
+
+/*
+  * # buildSearchQuery
+  *
+  * Internal proc that builds part of an SQL statement using a datum of search terms/parameters. It will then return
+  * a list of two objects: 1) the built SQL statement and 2) the assoc list of parameters that will accompany it in the query
+  * This should only ever be used to generate WHERE statements
+  *
+  * Arguments:
+  * * datum/library_user_data/search_terms - datum with parameters for what we want to query our DB for
+ */
+/datum/library_catalog/proc/buildSearchQuery(datum/library_user_data/search_terms)
```

Comment:
I don't know how to interpret this comment. This was the best way I could come up with the abstraction for building all the WHERE statements I need. It works quite well and it's insulated against SQL injections since at no point is the SQL statement ever built with user input, only the parameters take user input.